projects/igniteui-angular/src/lib/grid/grid-toolbar.component.ts

```diff
@@ -0,0 +1,114 @@
+import {
+    ChangeDetectorRef,
+    Component,
+    ContentChild,
+    Directive,
+    ElementRef,
+    HostBinding,
+    Input,
+    Optional,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+
+import { IgxButtonDirective } from '../directives/button/button.directive';
+import { IgxToggleDirective } from '../directives/toggle/toggle.directive';
+import { CsvFileTypes,
+         IgxCsvExporterOptions,
+         IgxCsvExporterService,
+         IgxExcelExporterOptions,
+         IgxExcelExporterService } from '../services/index';
+import { IgxGridAPIService } from './api.service';
+import { autoWire, IGridBus } from './grid.common';
+import { IgxGridComponent } from './grid.component';
+
+@Component({
+    selector: 'igx-grid-toolbar',
+    templateUrl: './grid-toolbar.component.html'
+})
+export class IgxGridToolbarComponent implements IGridBus {
+
+    @HostBinding('class.igx-grid-toolbar')
+    @Input()
+    public gridID: string;
+
+    @ViewChild(IgxToggleDirective, { read: IgxToggleDirective })
+    protected toggleDirective: IgxToggleDirective;
+
+    public get grid(): IgxGridComponent {
+        return this.gridAPI.get(this.gridID);
+    }
+
+    public get shouldShowExportButton(): boolean {
+        return (this.grid != null && (this.grid.exportExcel || this.grid.exportCsv));
+    }
+
+    public get shouldShowExportExcelButton(): boolean {
+        return this.grid.exportExcel;
+    }
+
+    public get shouldShowExportCsvButton(): boolean {
+        return this.grid.exportCsv;
+    }
```

Comment:
All the other fields except shouldShowExportExcelButton and shouldShowExportCsvButton have null checks for the grid. Shouldn't these two also have such check?